eth/downloader/downloader.go

```diff
@@ -1309,24 +1287,18 @@ func (d *Downloader) fetchHeaders(p *peerConnection, from uint64, height uint64)
 					log.Trace("getHeaders#downloadMoreHeaders", "from", from)
 					// If we're still skeleton filling fast sync, check pivot staleness
 					// before continuing to the next skeleton filling
-
-					// TODO(ponti): Re add the "moving" pivot, after changing the way we calculate the uptimeScore
-					// if skeleton && pivot > 0 {
-					// 	getNextPivot()
-					// } else {
-					getHeaders(from)
-					// }
+					if skeleton && pivot > 0 {
```

Comment:
My understanding of this check is that `pivot` is set from `pivotHeader` and `pivotHeader` is onlyset when performing fast sync, so this check is essentially checking whether we are still skeleton filling and performing a fast sync. But I'd just like to double check that with your understanding.

Reply:
Also why do we only update the pivot if we're still skeleton filling, I'd expect to want to update the pivot continuously regardless of where we are with respect to downloading skeleton headers or just headers?

Reply:
Good to know that you'll be in the fill skeleton stage until you reach the pivot, but I don't understand your point about the ton of problems. The pivot definitely gets changed after starting to download state, because state only exists for the 128 most recent blocks and there's no way you can download the whole state in that time, so the pivot has to be updated probably roughly every 10 minutes (because 128 blocks take 10.66 minutes)

Reply:
So from our chat, snap and fast sync are the only syncs that use a pivot, so `pivot > 0` is equivalent to `syncmode == FAST|SNAP`.

And we agreed that the pivot is constantly being updated, this is happening in `processFastSyncContent`

Reply:
So my current understanding is that in `fetchHeaders` the pivot is updated to be 64 blocks deep whenever it approaches 128 blocks deep, that pivot header is stored in `Downloader.pivotHeader` which is used by `processFastSyncContent, but for some reason `processFastSyncContent` also does some adjustment of the pivot to prevent staleness of the pivot.